src/utils/changeDurationToDate.ts

```diff
@@ -0,0 +1,37 @@
+/*-
+ *
+ * Hedera Token Holders List
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { DurationType } from '@/components/HoldersForm';
+
+export const changeDurationToDate = (duration: string | Date, durationType: DurationType): Date => {
+  if (typeof duration === 'string') {
+    const now = new Date();
+
+    switch (durationType) {
+      case 'days':
+        return new Date(now.setDate(now.getDate() - Number(duration)));
+      case 'weeks':
+        return new Date(now.setDate(now.getDate() - Number(duration) * 7));
+      case 'months':
+        return new Date(now.setMonth(now.getMonth() - Number(duration)));
```

Comment:
a month should be always treated as 30 days in our case

Reply:
Done